### PR TITLE
Run Project templates in separate test check

### PR DESCRIPTION
**Pull request by jkotalik**
For part of https://github.com/aspnet/AspNetCore-Internal/issues/2111

### CODE REVIEW

**Comment by jkotalik** [comment]
Longest running project is still Templates 😢 . But there are some wins we can get else where that I will follow up on. There are some things being built during build.cmd that shouldn't be, causing slowdown.

This is still a 20 minute improvement, so let's get this in ASAP.

**Review by ryanbrandenburg — APPROVED** [review]

**Review by analogrelay — APPROVED** [review]
Very nice!

File: .azure/pipelines/ci.yml [hunk 1]
```diff
@@ -313,6 +313,22 @@ jobs:
       path: artifacts/logs/
       publishOnError: true
 
+- template: jobs/default-build.yml
+  parameters:
+    condition: ne(variables['SkipTests'], 'true')
+    jobName: Windows_Templates_Test
+    jobDisplayName: "Test: Windows Server Template 2016 x64"
```
Comment (adrianwright109) [inline .azure/pipelines/ci.yml]: Is `Test: Windows Server Template 2016 x64` the best name for this Test?  Sounds like you've invented a new edition of Windows Server 2016 called Template. 

Better alternative might be:
`Test: Templates - Windows Server 2016 x64`

**Comment by jkotalik** [comment]
Tests failed due to abort.

**Review by dougbu — APPROVED** [review]
Nice. And, it doesn't conflict with #8877

File: .azure/pipelines/ci.yml [hunk 2]
```diff
@@ -313,6 +313,22 @@ jobs:
...
+- template: jobs/default-build.yml
+  parameters:
+    condition: ne(variables['SkipTests'], 'true')
+    jobName: Windows_Templates_Test
+    jobDisplayName: "Test: Templates - Windows Server 2016 x64"
+    agentOs: Windows
+    buildScript: ./src/ProjectTemplates/build.cmd
+    buildArgs: -ci -test -NoRestore -NoBuild -NoBuilddeps "/p:RunTemplateTests=true"
+    beforeBuild:
+    - powershell:  ./eng/scripts/cibuild.cmd
```
[the remark this answers is not in the document]
Comment (jkotalik) [reply .azure/pipelines/ci.yml]: I don't believe so. There is a new switch I will add to not build site extensions. But separate PR.

File: .azure/pipelines/ci.yml [hunk 3]
```diff
@@ -313,6 +313,22 @@ jobs:
...
+      displayName: Build Repo
+    artifacts:
+    - name: Windows_Test_Templates_Logs
```
[the remark this answers is not in the document]
Comment (jkotalik) [reply .azure/pipelines/ci.yml]: I'll do it in another PR. I'm impatient and want this ASAP to help the dev workflow and rather not wait another hour.

**Pull request merged**